src/jit/gentree.h

```diff
@@ -149,6 +149,18 @@ struct InlineCandidateInfo;
 
 typedef unsigned short AssertionIndex;
 
+//------------------------------------------------------------------------
+// GetAssertionIndex: return 1-based AssertionIndex from 0-based int index.
```

Comment:
I do not see why we can't make AssertionIndex  0-based with -1 as NO_ASSERTION_INDEX. It will allows us to delete this confusing conversions, that are easy to miss.